core/src/test/java/org/ehcache/EhcacheBasicRemoveAllEventsTest.java

```diff
@@ -0,0 +1,208 @@
+    final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+    final FakeStore fakeStore = new FakeStore(originalStoreContent);
+    this.store = spy(fakeStore);
+    final Set<String> keys = new LinkedHashSet<String>();
+    for (final String key : KEY_SET_A) {
+      keys.add(key);
+      if ("keyA2".equals(key)) {
+        keys.add(null);
+      }
+    }
+    final Ehcache<String, String> ehcache = this.getEhcache(null);
+    try {
+      ehcache.removeAll(keys);
+      fail();
+    } catch (NullPointerException e) {
+      // Expected
+    }
+    verify(cacheEventListener, never()).onEvent(Matchers.<CacheEvent<String, String>>any());
+    try {
+      ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+      fail();
+    } catch (UnsupportedOperationException e) {
+      //expected
+    }
+  }
+
+  @Test
+  public void testRemoveAll() throws Exception {
+    final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+    final FakeStore fakeStore = new FakeStore(originalStoreContent, Collections.singleton("keyA3"));
+    this.store = spy(fakeStore);
+    final Ehcache<String, String> ehcache = this.getEhcache(null);
+    final Set<String> contentUpdates = fanIn(KEY_SET_A, KEY_SET_C);
+    ehcache.removeAll(contentUpdates);
+    verify(cacheEventListener, times(5)).onEvent(argThat(isRemoved));
+    try {
+      ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+      fail();
+    } catch (UnsupportedOperationException e) {
+      //expected
+    }
+  }
+
+  @Test
+  public void testRemoveAllComplete() throws Exception {
+    final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+    final FakeStore fakeStore = new FakeStore(originalStoreContent);
+    this.store = spy(fakeStore);
+    final Ehcache<String, String> ehcache = this.getEhcache(null);
+
+    final Set<String> contentUpdates = fanIn(KEY_SET_A, KEY_SET_C);
+    ehcache.removeAll(contentUpdates);
+    verify(cacheEventListener, times(12)).onEvent(argThat(isRemoved));
+    try {
+      ehcache.getRuntimeConfiguration().deregisterCacheEventListener(cacheEventListener);
+      fail();
+    } catch (UnsupportedOperationException e) {
+      //expected
+    }
+  }
+
+  @Test
+  public void testRemoveAllStoreSomeOverlapCacheAccessExceptionBeforeWriterNoOverlapSomeFail() throws Exception {
+    final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+    final FakeStore fakeStore = new FakeStore(originalStoreContent);
+    this.store = spy(fakeStore);
+    doThrow(new CacheAccessException("")).when(this.store)
+        .bulkCompute(getAnyStringSet(), getAnyEntryIterableFunction());
+
+    final Map<String, String> originalWriterContent = getEntryMap(KEY_SET_B, KEY_SET_D);
+    final FakeCacheLoaderWriter fakeLoaderWriter = new FakeCacheLoaderWriter(originalWriterContent, KEY_SET_C);
```

Comment:
Cause we need the loader to perform some work while invoked before getting to the resilience strategy.